nltk/translate/bleu_score.py

```diff
@@ -166,18 +166,18 @@ def corpus_bleu(list_of_references, hypotheses, weights=(0.25, 0.25, 0.25, 0.25)
     if p_numerators[1] == 0:
         return 0
 
-    if not smoothing_function: # No smoothing, values remain as Fractions.
-        s = (w * math.log(p_i) for i, (w, p_i) in enumerate(zip(weights, p_n))
-             if p_i.numerator != 0)
-    else: # Smoothen the modified precision.
-        # Note: smoothing_function() may convert values into floats;
-        #       it tries to retain the Fraction object as much as the
-        #       smoothing method allows.
+    if smoothing_function:
+        # Smoothen the modified precision
         p_n = smoothing_function(p_n, references=references,
                                  hypothesis=hypothesis, hyp_len=hyp_len)
-        s = (w * math.log(p_i) for i, (w, p_i) in enumerate(zip(weights, p_n)))
 
-    return bp * math.exp(math.fsum(s))
+    s = (w * math.log(p_i) for i, (w, p_i) in enumerate(zip(weights, p_n)))
+
+    try:
+        return bp * math.exp(math.fsum(s))
+    except ValueError, e:
+        # ValueError means log(0.0). Return 0
+        return 0.0
 
```

Comment:
This try-except with `ValueError` should not happen. The purpose of the smoothing was such that the summation over the logs don't contain any 0 values.  Although the default smoothing is no-smoothing, and to avoid the `log(0.0)`, the if condition at https://github.com/nltk/nltk/blob/develop/nltk/translate/bleu_score.py#L170 would skip that order of ngram. So that will still not result in `ValueError`.

Could you write some test cases or examples that cause the `ValueError`?